### PR TITLE
Wire up shopify

### PR DESCRIPTION
**Pull request by ghost**
- Added cart creation
- Moved discount generation and cart creation to separate util files
- Added error messages if cart could not be created

closes [#98](https://github.com/LinumLabs/PsyDao-FE/issues/98) (last item on board, "wire-up")
closes [#100](https://github.com/LinumLabs/PsyDao-FE/issues/100) <-- error handling

### CODE REVIEW

File: utils/generateProductDiscount.ts [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+          products: {
+            productsToAdd: [`gid://shopify/Product/${SHOPIFY_PRODUCT_ID}`]
+          }
+        }
+      }
+    }
+  };
+
+  try {
+    // Note: OLD WAY (deprecated)
```
Comment (gaelicgod) [inline utils/generateProductDiscount.ts:107]: delete this please

Comment (EsterB12) [reply utils/generateProductDiscount.ts]: done!

File: utils/createShopifyCart.ts [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        headers: {
+          "Content-Type": "application/json",
+          "X-Shopify-Storefront-Access-Token":
+            STOREFRONT_API_PUBLIC_ACCESS_TOKEN
+        },
+        body: JSON.stringify({ query: cartCreateMutation, variables })
+      }
+    );
+
+    const resJSON = (await response.json()) as CartResponse;
```
Comment (gaelicgod) [inline utils/createShopifyCart.ts:71]: you're not checking `if resJSON.ok` here?

File: hooks/useRedirectToShopify.tsx [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  const redirectToShopify = async (ethAddress: Address | undefined) => {
+    if (!ethAddress) {
+      showCustomErrorToast("No wallet address provided.", width);
+      return;
+    }
+    const shopifyResponse = await fetch(
+      `/api/shopify?ethAddress=${ethAddress}`
+    );
+
+    const response = (await shopifyResponse.json()) as ShopifyResponse;
```
Comment (gaelicgod) [inline hooks/useRedirectToShopify.tsx:31]: same as the other one, is there a reason you're not checking .ok?

File: utils/createShopifyCart.ts (hunk 2 above)
Comment (EsterB12) [reply utils/createShopifyCart.ts:71]: Adding in both places! With regards to why it wasn't there, the absence of resJSON.ok didn't raise any flags when I tried to force errors and it slipped my mind :(

**Review by gaelicgod — APPROVED** [review]

**Pull request merged**